--- libraries/rtmidi/RtMidi.cpp	2019-10-03 21:04:13.010616000 +0900
+++ libraries/rtmidi/RtMidi_patched.cpp	2019-10-09 19:27:49.980757500 +0900
@@ -825,7 +825,11 @@
 
   portRef = MIDIGetSource( portNumber );
   nameRef = ConnectedEndpointName(portRef);
+#if defined( UNICODE ) || defined( _UNICODE )
+  CFStringGetCString( nameRef, name, sizeof(name), kCFStringEncodingUTF8);
+#else
   CFStringGetCString( nameRef, name, sizeof(name), CFStringGetSystemEncoding());
+#endif
   CFRelease( nameRef );
 
   return stringName = name;
@@ -897,7 +901,11 @@
 
   portRef = MIDIGetDestination( portNumber );
   nameRef = ConnectedEndpointName(portRef);
+#if defined( UNICODE ) || defined( _UNICODE )
+  CFStringGetCString( nameRef, name, sizeof(name), kCFStringEncodingUTF8);
+#else
   CFStringGetCString( nameRef, name, sizeof(name), CFStringGetSystemEncoding());
+#endif
   CFRelease( nameRef );
   
   return stringName = name;
